rancher2/resource_rancher2_app_v2.go

```diff
@@ -49,11 +50,16 @@ func resourceRancher2AppV2Create(d *schema.ResourceData, meta interface{}) error
 	}
 	d.Set("cluster_name", cluster.Name)
 
-	systemDefaultRegistry, err := meta.(*Config).GetSettingV2ByID(appV2DefaultRegistryID)
+	globalSystemDefaultRegistry, err := meta.(*Config).GetSettingV2ByID(appV2DefaultRegistryID)
 	if err != nil {
 		return err
 	}
-	d.Set("system_default_registry", systemDefaultRegistry.Value)
+
+	if systemDefaultRegistry == "" {
+		d.Set("system_default_registry", globalSystemDefaultRegistry.Value)
+	} else {
+		d.Set("system_default_registry", systemDefaultRegistry)
```

Comment:
@nicholasSUSE I think the else condition is not necessary since line 40 already has `system_default_registry` set by the user.